k2/python/k2/fsa.py

```diff
@@ -259,6 +259,28 @@ def __str__(self) -> str:
         '''
         return self.to_str(openfst=False)
 
+    def get_filler(self, attribute_name: str) -> Union[int, float]:
+        '''
+        Return the filler value associated with attribute names.  This is
+        0 unless otherwise specified, but you can override this by
+        for example, doing
+            fsa.foo_filler = -1
+        which will mean the "filler" for attribute fsa.foo is -1; and
+        this will get propagated when you do FSA operations, like any
+        other non-tensor attribute.  The filler is the value that means
+        "nothing is here" (like epsilon).
+        Caution: you should use a value that is castable to float and back
+        to integer without loss of precision, because currently the
+        `default_value` parameter of `index_select` in ./ops.py is a float.
+        '''
+        try:
```

Comment:
How about
```python
assert attribute_name != 'aux_labels', 'you cannot set the filler for aux_labels'

ans = getattr(self, attribute_name + '_filler', 0)

return ans
```

Reply:
This doesn't work because we still *get* the filler for 'aux_labels', we just expect it to not be set.  Will change it, though, to also allow it to be set to 0.

Reply:
In that case, 
```python
ans = getattr(self, attribute_name + '_filler', 0) 
assert attribute_name != 'aux_labels' or ans == 0, 'you cannot set the filler for aux_labels' 
return ans
```
My intention is that

- (1) You don't need to use `try ... except` here. `getattr` accepts a default value

- (2) I usually see people using `assert some_expr, "some message"` in Python.
People in C/C++ use `assert(some_expr && "some message");`